frontend/src/components/Folders/Folder.jsx

```diff
@@ -1,12 +1,121 @@
+import { useEffect, useState } from 'react'
 import './Folder.css'
+import { doc, updateDoc, arrayUnion, getDoc, setDoc, collection, addDoc } from 'firebase/firestore';
+import { db } from '../../../../backend/firebase/firebase.config';
+import { useAuth } from '../auth/auth.jsx';
 
+const Folder = ({ folderName }) => {
+    const { currentUser } = useAuth();
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [transcripts, setTranscripts] = useState([]);
+    const [isTranscriptOpen, setIsTranscriptOpen] = useState(false);
+    const [currentTranscriptName, setCurrentTranscriptName] = useState("");
+    const [transcriptData, setTranscriptData] = useState([]);
 
 
-const Folder = ({folderName}) => {
+    const getTranscripts = async () => {
+        const docRef = doc(db, "users", currentUser.uid)
+        const userDoc = await getDoc(docRef);
+        if (!userDoc.exists()) {
```

Comment:
Just wondering, should we create this in case it doesn't exist, since we are calling `userDoc.data()` below?